### ISSUE TITLE
Error when using predict on a large rfsrc classification model

### ISSUE THREAD

**Issue by gse-cc-git** [kogalur/randomForestSRC#8]
Hello,
I am trying to use `randomForestSRC` on a classification problem.

Below the error message I get:
```
all scheduled cores encountered errors in user codeError in as.data.frame.default(x[[i]], optional = TRUE, stringsAsFactors = stringsAsFactors) : 
  cannot coerce class ‘"try-error"’ to a data.frame
```

I suspected the usage of a data.table object, but the following code worked fine.

```
data(iris)

dt_iris <- as.data.table(iris)
iris_modl_rfsrc      <- rfsrc(Species ~., data = dt_iris)
iris_pred_rfsrc.pred <- predict(object=iris_modl_rfsrc,newdata=dt_iris[,.SD,.SDcols=-"Species"])
```

Thanks for your help.

My sessionInfo:
```
version  R version 3.5.0 (2018-04-23)
 system   x86_64, linux-gnu           
 ui       RStudio (1.1.447)           
 language (EN)  
Packages --------------------------------------------------------------------------------------------------------------------
 package         * version  date       source        
 assertthat        0.2.0    2017-04-11 CRAN (R 3.5.0)
 backports         1.1.2    2017-12-13 CRAN (R 3.5.0)
 base            * 3.5.0    2018-04-25 local         
 base64enc         0.1-3    2015-07-28 CRAN (R 3.5.0)
 bindr             0.1.1    2018-03-13 CRAN (R 3.5.0)
 bindrcpp          0.2.2    2018-03-29 CRAN (R 3.5.0)
 colorspace        1.3-2    2016-12-14 CRAN (R 3.5.0)
 compiler          3.5.0    2018-04-25 local         
 data.table      * 1.10.4-3 2017-10-27 CRAN (R 3.5.0)
 datasets        * 3.5.0    2018-04-25 local         
 devtools          1.13.5   2018-02-18 CRAN (R 3.5.0)
 digest            0.6.15   2018-01-28 CRAN (R 3.5.0)
 dplyr             0.7.4    2017-09-28 CRAN (R 3.5.0)
 DT                0.4      2018-01-30 CRAN (R 3.5.0)
 evaluate          0.10.1   2017-06-24 CRAN (R 3.5.0)
 foreign           0.8-70   2018-04-23 CRAN (R 3.5.0)
 ggplot2           2.2.1    2016-12-30 CRAN (R 3.5.0)
 glue              1.2.0    2017-10-29 CRAN (R 3.5.0)
 graphics        * 3.5.0    2018-04-25 local         
 grDevices       * 3.5.0    2018-04-25 local         
 grid              3.5.0    2018-04-25 local         
 gtable            0.2.0    2016-02-26 CRAN (R 3.5.0)
 htmltools         0.3.6    2017-04-28 CRAN (R 3.5.0)
 htmlwidgets       1.2      2018-04-19 CRAN (R 3.5.0)
 httr              1.3.1    2017-08-20 CRAN (R 3.5.0)
 jsonlite          1.5      2017-06-01 CRAN (R 3.5.0)
 knitr             1.20     2018-02-20 CRAN (R 3.5.0)
 lattice           0.20-35  2017-03-25 CRAN (R 3.5.0)
 lazyeval          0.2.1    2017-10-29 CRAN (R 3.5.0)
 magrittr          1.5      2014-11-22 CRAN (R 3.5.0)
 maptools          0.9-2    2017-03-25 CRAN (R 3.5.0)
 memoise           1.1.0    2017-04-21 CRAN (R 3.5.0)
 methods         * 3.5.0    2018-04-25 local         
 munsell           0.4.3    2016-02-13 CRAN (R 3.5.0)
 parallel          3.5.0    2018-04-25 local         
 pillar            1.2.1    2018-02-27 CRAN (R 3.5.0)
 pkgconfig         2.0.1    2017-03-21 CRAN (R 3.5.0)
 plotly            4.7.1    2017-07-29 CRAN (R 3.5.0)
 plyr              1.8.4    2016-06-08 CRAN (R 3.5.0)
 purrr             0.2.4    2017-10-18 CRAN (R 3.5.0)
 R6                2.2.2    2017-06-17 CRAN (R 3.5.0)
 randomForestSRC * 2.6.0    2018-05-02 CRAN (R 3.5.0)
 Rcpp              0.12.16  2018-03-13 CRAN (R 3.5.0)
 rgeos             0.3-26   2017-10-31 CRAN (R 3.5.0)
 rlang             0.2.0    2018-02-20 CRAN (R 3.5.0)
 rmarkdown         1.9      2018-03-01 CRAN (R 3.5.0)
 rprojroot         1.3-2    2018-01-03 CRAN (R 3.5.0)
 scales            0.5.0    2017-08-24 CRAN (R 3.5.0)
 sp                1.2-7    2018-01-19 CRAN (R 3.5.0)
 splitstackshape * 1.4.4    2018-03-29 CRAN (R 3.5.0)
 stats           * 3.5.0    2018-04-25 local         
 stringi           1.1.7    2018-03-12 CRAN (R 3.5.0)
 stringr           1.3.0    2018-02-19 CRAN (R 3.5.0)
 tibble            1.4.2    2018-01-22 CRAN (R 3.5.0)
 tidyr             0.8.0    2018-01-29 CRAN (R 3.5.0)
 tools             3.5.0    2018-04-25 local         
 utils           * 3.5.0    2018-04-25 local         
 viridisLite       0.3.0    2018-02-01 CRAN (R 3.5.0)
 withr             2.1.2    2018-03-15 CRAN (R 3.5.0)
 yaml              2.1.18   2018-03-08 CRAN (R 3.5.0
```

**Comment by kogalur** [comment]
Please supply us with a minimal code example with which we can reproduce and investigate your issue.

**Comment by gse-cc-git** [comment]
Hello,
I use large files, and did not succeed in find the problematic dataset features so that I could produce a minimal reproducible example. I must conclude that my issue was due to discrepancies between formatting of the training and test datasets.

Now that I regenerated the datasets on the same basis I am facing a new issue: the output of the predicted object gives a surprising equiprobable distribution (I mean tests cases are almost evenly dispatched among the class to predict labels).

table(prediction_output$class)

![selection_118](https://user-images.githubusercontent.com/12275706/40002594-7e584d2c-5791-11e8-80ac-1d5c99b9f4b9.png)


Shall I close this issue and open a new one ?
Thanks a lot for your work and  for your time.
